nltk/sentiment/vader.py

```diff
@@ -433,30 +448,76 @@ def _least_check(self, valence, words_and_emoticons, i):
+            valence = valence * self.constants.N_SCALAR
         return valence
 
-    def _but_check(self, words_and_emoticons, sentiments):
-        # check for modification in sentiment due to contrastive conjunction 'but'
-        if "but" in words_and_emoticons or "BUT" in words_and_emoticons:
-            try:
-                bi = words_and_emoticons.index("but")
-            except ValueError:
-                bi = words_and_emoticons.index("BUT")
-            for sentiment in sentiments:
-                si = sentiments.index(sentiment)
-                if si < bi:
-                    sentiments.pop(si)
-                    sentiments.insert(si, sentiment * 0.5)
-                elif si > bi:
-                    sentiments.pop(si)
-                    sentiments.insert(si, sentiment * 1.5)
+    def _but_check(self, words_and_emoticons, sentiments, extended=False):
+        if extended:
+            # Extended version from
+            # https://medium.com/@malavika.suresh0794/vader-customizing-the-library-71d9e8ed6eda
+            # check for modification in sentiment due to contrastive conjunctions
+            words_and_emoticons_lower = [str(w).lower() for w in words_and_emoticons]
+            cc_list = ['but', 'however', 'except']
+            bi = 0
+            for cc in cc_list:
+                if cc in words_and_emoticons_lower:
+                    bi = words_and_emoticons_lower.index(cc)
+                    for si,sentiment in enumerate(sentiments):
+                        if si < bi:
+                            sentiments[si] = sentiment * 0.5
+                        elif si > bi:
+                            sentiments[si] = sentiment * 1.5
+            # Future work:
+            # 1.Consider usage of though/although/even though
+            return sentiments
+        else:
+            # Original Vader code.
+            # check for modification in sentiment due to contrastive conjunction 'but'
+            if "but" in words_and_emoticons or "BUT" in words_and_emoticons:
+                try:
+                    bi = words_and_emoticons.index("but")
+                except ValueError:
+                    bi = words_and_emoticons.index("BUT")
+                for sentiment in sentiments:
+                    si = sentiments.index(sentiment)
+                    if si < bi:
+                        sentiments.pop(si)
+                        sentiments.insert(si, sentiment * 0.5)
+                    elif si > bi:
+                        sentiments.pop(si)
+                        sentiments.insert(si, sentiment * 1.5)
+            return sentiments
+
+    def _only_if_check(self, words_and_emoticons, sentiments):
+        words_and_emoticons_lower = [str(w).lower() for w in words_and_emoticons]
+        check = 'only'
+        if check in words_and_emoticons_lower:
+            i = words_and_emoticons_lower.index(check)
+            if len(words_and_emoticons)>i+1 and 'if' == words_and_emoticons_lower[i+1]:
```

Comment:
On a closer look, this is trying to find the index of the Multi-word expression "only if". Maybe using ngrams and index of ngrams would work.

Reply:
E.g. 

```python
from nltk.util import ngrams

x = "It was smooth sailing in spite of the trade tension".split()

for idx, ng in enumerate(ngrams(x, 3)):
    if ng == ('in', 'spite', 'of'):
        print(x[:idx])
```

Reply:
Yep, you're right.